### DISCUSSION TITLE
Missing feature: kill switch

### DISCUSSION THREAD

**Discussion by ghost** [oxen-io/lokinet#1886] [General]
need kill switch most vpn have it

**Comment by majestrate** [comment]
Due to windows firewall not being able to do it's job most users refer to a competent firewall that does its job as a kill switch. We are adding this to the win32 port (slowly). This task is very tedious and has been dragging out for longer than i'd like. During that (mis)adventure it has become painfully noticeable to me that VPN and other network related APIs on windows (if they exist to begin with) are second class citizens. This is not shocking given MS's perception that the internet was a fad when the development of these APIs back in the 90s and mid 2000s was occurring. We moved the win32 VPN platform code to use wireguard's wintun so that should make the code a lot less hairy. We should have a very basic version of this in the v0.9.9 tag.
